### ISSUE TITLE
Integration with Create React App by default, including types support for typescript

### ISSUE THREAD

**Issue by ayoseturru** [facebook/stylex#482]
#### Description:
I would like to propose adding native support for StyleX in Create React App (CRA) by default. This would eliminate the need for users to eject or maintain a custom babel config file solely for integrating StyleX into their projects.

#### Benefits:
1. **Streamlined Development**: Users can seamlessly integrate StyleX into their Create React App projects without the need for additional configuration steps.
2. **Accessibility**: Making StyleX available by default would promote its usage and encourage developers to explore its features, thereby potentially increasing adoption.
3. **Improved Developer Experience**: Simplifying the process of integrating StyleX can enhance the overall developer experience and reduce friction when working on React projects.

#### Implementation Details:
- Modify the Create React App template to include support for StyleX out of the box.
- Ensure that necessary dependencies and configurations are included to enable StyleX usage without additional setup steps.
- Provide documentation and examples to guide users on how to utilize StyleX within Create React App projects.

#### Additional Considerations:
- Compatibility with existing Create React App features and functionalities.
- Potential impact on build size, performance, and other aspects of project development.

#### Conclusion:
Integrating StyleX into Create React App by default would simplify the process of using this library and encourage its adoption among React developers. This feature enhancement could contribute to a more streamlined development experience and facilitate the creation of stylish and responsive web applications.

**Comment by necolas** [comment]
This request is probably best to make in the CRA repo

**Comment by nmn** [comment]
#34 covers the current issues with Webpack. As we fix that, CRA should be unblocked. Closing as a duplicate of that issue.
